patches/tModLoader/Terraria/ModLoader/ItemLoader.cs

```diff
@@ -1431,89 +1431,107 @@ public static bool CanStackInWorld(Item increase, Item decrease)
 	private static HookList HookOnStack = AddHook<Action<Item, Item, int>>(g => g.OnStack);
 
 	/// <summary>
-	/// Short-circuits if <see cref="CanStack"/> hooks don't return true.
-	/// <br/> Returns the result of the initial check.
-	/// <br/>
-	/// <br/> <inheritdoc cref="StackItems"/>
+	/// Stacks <paramref name="source"/> onto <paramref name="destination"/> if CanStack permits the transfer
 	/// </summary>
-	/// <param name="increase">Item where the stack is being increased.</param>
-	/// <param name="decrease">Item where the stack is being reduced.</param>
-	/// <param name="numTransfered">Amount to be transfered </param>
-	/// <param name="infiniteSource">The final stack of item2</param>
-	public static bool TryStackItems(Item increase, Item decrease, out int numTransfered, bool infiniteSource = false)
+	/// <param name="destination">The item instance that <paramref name="source"/> will attempt to stack onto</param>
+	/// <param name="source">The item instance being stacked onto <paramref name="destination"/></param>
+	/// <param name="numTransferred">The quanity of <paramref name="source"/> that was transferred to <paramref name="destination"/></param>
+	/// <param name="infiniteSource">Whether <paramref name="source"/>.stack should be decreased</param>
+	public static bool TryStackItems(Item destination, Item source, out int numTransferred, bool infiniteSource = false)
 	{
-		numTransfered = 0;
+		numTransferred = 0;
 
-		if (!CanStack(increase, decrease))
+		if (!CanStack(destination, source))
 			return false;
 
-		StackItems(increase, decrease, out numTransfered, infiniteSource);
+		StackItems(destination, source, out numTransferred, infiniteSource);
 
 		return true;
 	}
 
 	/// <summary>
-	/// Stacks <paramref name="increase"/> and <paramref name="decrease"/> items.
-	/// <br/> Calls all <see cref="GlobalItem.OnStack"/> and <see cref="ModItem.OnStack"/> hooks if <paramref name="increase"/>.stack is less than <paramref name="increase"/>.maxStack.
+	/// Attempts to stack <paramref name="destination"/> onto <paramref name="source"/>
```

Comment:
`Attempts` is misleading. Need an explicit warning not to call this unless `CanStack` has previously been checked, and to see `TryStackItems` instead.